docs/translations.md

```diff
@@ -0,0 +1,270 @@
+
+
+To facilitate getting the locale value in nested SSCs, it is recommended to wrap the top-level SSCs with the Higher-Order-Component (HOC) *withTranslations* that is defined in *src/core/server/locale.ts*
+
+```ts
+export default withTranslations<HomePageProps>(function HomePage() {
+    const t = useTranslations();
+    ///...
+}
+```
+
+### Accessing Translations (async components)
+
+NextJS allows us to make DB and other async calls that require the component to be defined as *async*. Due to technical limitations, it is not possible to use the hook *useTranslations* under this circumstance. The solution to this problem is to make use of *getTranslations* defined in *src/core/server/getTranslations.ts*
+
+```ts
+export default withTranslations<PageProps>(async function AllProductsPage() {
+    const t = await getTranslations();
+    //...
+}
+```
+
+### Saleor API Translations
+
+The Saleor API can provide translations for some entity properties. The translations object may be null if there are no values for the specific data requested. 
+
+An example GraphQL query for requesting **Products**
+```gql
+fragment GalleryProductFragment on Product {
+  id
+  slug
+  name
+  description
+  translation(languageCode: $languageCode) {
+    name
+    description
+  }
+}
+```
+
+The *$languageCode* value is required to be an all-caps ISO format language code (e.g. EN_US). Although there is no standard for defining the *locale* value in the [localeConfig](#supported-languages), we have adopted a \<language>-\<country-code> format (e.g. en-us). A helper function was defined to translate this format to the required Saleor API standard.
+
+*src/core/server/locale.ts*
+```ts
+function getLanguageCode(locale: Locale): ILanguageCodeEnum
+```
+
+The product data is defined in the *Saleor Dashboard* with a base language which will be the values returned by the *name* and *description* properties of **Product**. If the product has translation values for the requested locale, then those will be part of the *translation* property of the returned data. This make it possible to use a default fallback language in case translations are not available, but as-is, will require some complex logic and condition to identify existing translations for each property. 
+
+A proposed solution to this problem is to make use of hooks that will create a new Saleor entity object that will prefer to use the translation values if available, and fallback to the default value if translations are not available. There are experimental implementations using this approach for the **Product** and **Menu** entities.
+
+*src/core/client/useProductTranslation.tsx*
+```ts
+function useProductTranslation(product: Translatable)
+```
+
+*src/core/server/getMenuTranslation.ts*
```

Comment:
I don't know that this is relevant to these docs (which are great), but does the `server/` vs. `client/` folder separation make sense long term? What if some set of modules are meant as companions spanning the client and server?